From 31dcd4e2d08108733685f35375e31f88583591d5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Armin=20Krezovi=C4=87?= <krezovic.armin@gmail.com>
Date: Fri, 29 Jul 2016 13:26:22 +0200
Subject: [PATCH 03/94] compositor-drm: Do not return an error when no
 connectors are configured
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Returning an error when there are no connectors results in
weston terminating after that. That's not expected when
trying to get weston to start with zero drm outputs.

Signed-off-by: Armin KrezoviÄ‡ <krezovic.armin@gmail.com>
Reviewed-by: Pekka Paalanen <pekka.paalanen@collabora.co.uk>
---
 src/compositor-drm.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/compositor-drm.c b/src/compositor-drm.c
index 893877d..64e820c 100644
--- a/src/compositor-drm.c
+++ b/src/compositor-drm.c
@@ -2569,11 +2569,8 @@ create_outputs(struct drm_backend *b, uint32_t option_connector,
 		drmModeFreeConnector(connector);
 	}
 
-	if (wl_list_empty(&b->compositor->output_list)) {
+	if (wl_list_empty(&b->compositor->output_list))
 		weston_log("No currently active connector found.\n");
-		drmModeFreeResources(resources);
-		return -1;
-	}
 
 	drmModeFreeResources(resources);
 
-- 
1.9.1

